requirements.txt

```diff
@@ -1,265 +1,62 @@
-nbformat==5.7.0
-nbval==0.9.6
-ndg-httpsclient==0.5.1
-nest-asyncio==1.5.6
-networkx==2.5.1
-notebook==6.4.12
-numpy==1.22.0
+networkx==3.1
+numpy==1.25.1
 oauthlib==3.2.2
-opencensus==0.11.0
+opencensus==0.11.2
 opencensus-context==0.1.3
-opencensus-ext-azure==1.1.7
-opt-einsum==3.3.0
-packaging==21.3
-pandas==1.5.0
-pandocfilters==1.5.0
-paramiko==2.11.0
-parso==0.8.3
-partd==1.3.0
-pathspec==0.10.1
-pexpect==4.8.0
-pickleshare==0.7.5
-pillow==9.3.0
-pkgutil-resolve-name==1.3.10
-platformdirs==2.5.2
-pluggy==1.0.0
-portalocker==2.5.1
-prettytable==3.4.1
-prometheus-client==0.14.1
-prompt-toolkit==3.0.31
-protobuf==3.20.3
-psutil==5.9.2
-ptyprocess==0.7.0
-pure_eval==0.2.2
-py==1.11.0
-py-cpuinfo==9.0.0
-py4j==0.10.9.7
-pyarrow==6.0.1
-pyasn1==0.4.8
-pyasn1-modules==0.2.8
+opencensus-ext-azure==1.1.9
+portalocker==2.7.0
+protobuf==4.23.4
+psutil==5.9.5
+pyasn1==0.5.0
+pyasn1-modules==0.3.0
 pycparser==2.21
-Pygments==2.13.0
-PyJWT==2.5.0
-PyNaCl==1.5.0
-pyOpenSSL==22.1.0
-pyparsing==3.0.9
-pyrsistent==0.18.1
-pyspark==3.3.0
-pytest==7.1.3
-pytest-benchmark==4.0.0
-pytest-forked==1.4.0
-pytest-rerunfailures==10.2
-pytest-xdist==2.5.0
+PyJWT==2.8.0
+pyparsing==3.1.0
 python-dateutil==2.8.2
-pytz==2022.4
-pywin32==304;sys_platform == 'win32'
-PyYAML==6.0
-pyzmq==24.0.1
-ray[tune]==2.2.0
-rdflib==6.2.0
-regex==2022.9.13
 requests==2.31.0
 requests-oauthlib==1.3.1
 rsa==4.9
-ruamel.yaml==0.17.21
-ruamel.yaml.clib==0.2.6
-s3transfer==0.6.0
-sacremoses==0.0.53
-safetensors==0.3.1
-scikit-learn==1.1.2
-scipy==1.10.0
-SecretStorage==3.3.3
-Send2Trash==1.8.0
-sentencepiece==0.1.97
-setuptools==65.5.1
-shrike==1.31.10
+scikit-learn==1.3.0
+scipy==1.11.1
 six==1.16.0
-snowballstemmer==2.2.0
-sortedcontainers==2.4.0
-sphinx==5.2.3
-sphinx-copybutton==0.5.0
-sphinxcontrib-applehelp==1.0.2
-sphinxcontrib-devhelp==1.0.2
-sphinxcontrib-htmlhelp==2.0.0
-sphinxcontrib-jsmath==1.0.1
-sphinxcontrib-qthelp==1.0.3
-sphinxcontrib-serializinghtml==1.1.5
-sphinxcontrib-websupport==1.2.4
-stack_data==0.5.1
-tabulate==0.8.10
-tblib==1.7.0
-tenacity==8.2.1
-tensorboard==2.10.1
-tensorboard-data-server==0.6.1
-tensorboard-plugin-wit==1.8.1
-tensorboardX==2.5.1
-tensorflow==2.11.1
-tensorflow-addons==0.18.0
-tensorflow-estimator==2.10.0
-tensorflow-intel==2.11.1
-tensorflow-io-gcs-filesystem==0.27.0
-termcolor==2.0.1
-terminado==0.16.0
-testpath==0.6.0
-threadpoolctl==3.1.0
-thrift==0.16.0
-tokenizers==0.13.1
-toml==0.10.2
-tomli==2.0.1
-toolz==0.12.0
-torch==1.12.1
-torch-cluster==1.6.0
-torch-scatter==2.0.9
-torch-sparse==0.6.15
-torch_geometric==2.1.0
-tornado==6.3.2
-tqdm==4.64.1
-traitlets==5.4.0
-transformers==4.30.2
-typeguard==2.13.3
-typing-extensions==4.4.0
-urllib3==1.26.12
-virtualenv==20.16.5
-wcwidth==0.2.5
-webencodings==0.5.1
-websocket-client==1.4.1
-Werkzeug==2.2.3
-wheel==0.38.1
-wrapt==1.14.1
-yacs==0.1.8
-yarl==1.8.1
-zict==2.2.0
-zipp==3.9.0
+sympy==1.12
+tenacity==8.2.2
+threadpoolctl==3.2.0
+--extra-index-url https://download.pytorch.org/whl/cpu
```

Comment:
This belongs not in file or at top of file maybe.

Reply:
why? It is needed for torch package, that is jut below it.

Reply:
Then the extra_index_url at the top of the file used for torch geometric should be moved down next to it also.